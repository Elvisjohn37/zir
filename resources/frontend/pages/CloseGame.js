import React, { useEffect } from 'react';
import styles from './CloseGame.module';
import Spinner from 'frontend/components/Spinner';
import { FormattedMessage } from 'react-intl';
import { LangDataProvider } from 'frontend/components/Language';

export default function CloseGame() {
	useEffect(() => {
		parent.postMessage('closeGame', '*');
	}, []);

	return (
		<LangDataProvider category="gamewindow">
			<div className={styles.loadingMessage}>
				<Spinner color="primary" />
				<FormattedMessage id="returningToLobby" />
			</div>
		</LangDataProvider>
	);
}
