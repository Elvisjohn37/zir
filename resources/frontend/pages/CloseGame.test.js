import React from 'react';
import { render } from '@testing-library/react';
import CloseGame from './CloseGame';

jest.mock('frontend/components/Spinner');
jest.mock('react-intl');
jest.mock('frontend/components/Language');

let postMessageFn = jest.fn();
window.parent.postMessage = postMessageFn;

describe('</CloseGame>', () => {
	it('should match snapshot', () => {
		let { asFragment } = render(<CloseGame />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('should have called postMessage with closeGame param on mount', () => {
		render(<CloseGame />);
		expect(postMessageFn).toHaveBeenCalledWith('closeGame', '*');
	});
});
