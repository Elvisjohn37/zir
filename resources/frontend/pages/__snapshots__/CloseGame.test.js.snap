// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`</CloseGame> should match snapshot 1`] = `
<DocumentFragment>
  <div
    category="gamewindow"
    data-testid="LangDataProvider"
  >
    <div>
      gamewindow
    </div>
    <div
      class="loadingMessage"
    >
      <div
        color="primary"
        data-testid="Spinner"
      >
        <div>
          Loading Spinner
        </div>
      </div>
      <div
        data-testid="FormattedMessage"
        id="returningToLobby"
      >
        <div>
          returningToLobby
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
